tests/button.spec.ts

```diff
@@ -0,0 +1,95 @@
+  await expect(successButton.locator('button')).toHaveClass(/text-white/)
+})
+
+test('Renders a button in the style of danger if the variant is specified as danger', async ({ page }) => {
+  const dangerButton = page.getByTestId('danger')
+  await expect(dangerButton.locator('button').first()).toBeVisible()
+  await expect(dangerButton.locator('button').first()).toHaveClass(/bg-danger-fg/)
+  await expect(dangerButton.locator('button').first()).toHaveClass(/border-danger-fg/)
+  await expect(dangerButton.locator('button').first()).toHaveClass(/text-white/)
+})
+
+test('Renders a button in the style of outline danger if the variant is specified as outline danger', async ({ page }) => {
+  const outlineDangerButton = page.getByTestId('outline-danger')
+  await expect(outlineDangerButton.locator('button')).toBeVisible()
+  await expect(outlineDangerButton.locator('button')).toHaveClass(/bg-danger-bg/)
+  await expect(outlineDangerButton.locator('button')).toHaveClass(/border-danger-bg/)
+  await expect(outlineDangerButton.locator('button')).toHaveClass(/text-danger-fg/)
+})
+
+test('Renders a button as disabled if the button has a disabled attribute of true', async ({ page }) => {
+  const disabledButton = page.getByRole('button', { name: 'Disabled Label' })
+  await expect(disabledButton).toBeVisible()
+  await expect(disabledButton).toBeDisabled()
+})
+
+test('Renders a tooltip over the button if a tooltip attribute is specified', async ({ page }) => {
+  const tooltipButton = page.getByRole('button', { name: 'Tooltip Label' })
+  const tooltipText = page.getByRole('tooltip', { name: 'This is the message for the tooltip'})
+  await expect(tooltipButton).toBeVisible()
+  await tooltipButton.hover()
+  await expect(tooltipText).toBeVisible()
+  await expect(tooltipText).toHaveText('This is the message for the tooltip')
+})
+
+test('Renders an icon within the button/next to the label if an icon attribute is specified', async ({ page }) => {
+  const iconWithinButton = page.getByRole('button', { name: 'Icon Within Button Label' })
+  await expect(iconWithinButton).toBeVisible()
+  await expect(iconWithinButton).toContainText('Label')
+  await expect(iconWithinButton.locator('i')).toHaveClass(/icon-refresh/)
+  await expect(iconWithinButton).toHaveClass(/border-gray-9/)
+})
+
+test('Renders the button as an icon if an icon attribute is specified and the button variant is icon', async ({ page }) => {
+  const iconAsButton = page.getByRole('button', { name: 'Icon As Button' })
+  await expect(iconAsButton).toBeVisible()
+  await expect(iconAsButton).not.toContainText('Icon As Button')
+  await expect(iconAsButton.locator('i')).toHaveClass(/icon-refresh/)
+  await expect(iconAsButton).not.toHaveClass(/border-gray-9/)
+})
+
+test('Tests that the button is clickable if not disabled', async ({ page }) => {
+  const clickMeButton = page.getByRole('button', { name: 'Click Me' })
+  await clickMeButton.isEnabled()
+  await clickMeButton.click()
```

Comment:
Using a .click without an assertion is sufficient in this clickable test given when using .click, playwright checks if the element is clickable by confirming the element is:

- Attached to the DOM
- Visible
- Stable
- Receives events
- Is enabled

If .click doesn't pass those checks, this test will fail.